test/pipelinerun_test.go

```diff
@@ -45,6 +45,146 @@ var (
 	pipelineRunTimeout = 10 * time.Minute
 )
 
+func TestPipelineRunWithRetry(t *testing.T) {
+	t.Parallel()
+	type tests struct {
+		name                   string
+		testSetup              func(t *testing.T, c *clients, namespace string, index int)
+		expectedTaskRuns       []string
+		expectedNumberOfEvents int
+		pipelineRunFunc        func(int, string) *v1alpha1.PipelineRun
+		expectedRetries        int
+	}
+
+	tds := []tests{{
+		name:            "timeout with a retry",
+		expectedRetries: 1,
+		testSetup: func(t *testing.T, c *clients, namespace string, index int) {
+			t.Helper()
+			if _, err := c.KubeClient.Kube.CoreV1().Secrets(namespace).Create(getPipelineRunSecret(index, namespace)); err != nil {
+				t.Fatalf("Failed to create secret `%s`: %s", getName(secretName, index), err)
+			}
+
+			if _, err := c.KubeClient.Kube.CoreV1().ServiceAccounts(namespace).Create(getPipelineRunServiceAccount(index, namespace)); err != nil {
+				t.Fatalf("Failed to create SA `%s`: %s", getName(saName, index), err)
+			}
+
+			task := tb.Task(getName(taskName, index), namespace, tb.TaskSpec(
+				// Reference build: https://github.com/knative/build/tree/master/test/docker-basic
+				tb.Step("config-docker", "gcr.io/cloud-builders/docker",
+					tb.Command("docker"),
+					tb.Args("pull", "gcr.io/build-crd-testing/secret-sauce"),
+					tb.VolumeMount("docker-socket", "/var/run/docker.sock"),
+				),
+				tb.TaskVolume("docker-socket", tb.VolumeSource(corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: "/var/run/docker.sock",
+						Type: newHostPathType(string(corev1.HostPathSocket)),
+					},
+				})),
+			))
```

Comment:
if im understanding this correctly, the goal is for the task to fail, is that right? how about having the step do something simpler, that immediately fails, e.g. something like `exit 1`?